### ISSUE TITLE
items under "your music" times out with asynchronous error

### ISSUE THREAD

**Issue by brumle80** [fuzeman/Spotify2.bundle#39]
this happens on plexweb, both local and plex.tv, 
haven't tested iphone app, but happens with plexpass app on android.

error message when trying to open playlists, starred & albums:
"Asynchronous callback timed out before returning a result"

clicking artists works though :+1: 

Plex server is Plexpass Version 0.9.9.13 on Fedora 19
Spotify.bundle is latest release.

debug log:

http://pastebin.com/StxaPqht

**Comment by brumle80** [comment]
Plex server is Plexpass Version 0.9.9.13 on Fedora 19
Spotify.bundle is latest release.

**Comment by fuzeman** [comment]
There is a fix coming for albums, it looks like the proper media objects for them in plex have been broken with a recent PMS update.

Looking at your log file, there seems to be another issue as a result of having local media, should be fairly simple to fix this.

I'll let you know when the update (which fixes these issues) has been released.

**Comment by brumle80** [comment]
thanks :D

**Comment by fuzeman** [comment]
After looking at this issue more it doesn't look like a problem with local media in starred/playlist containers.

---

To try figure out what's happening, could you grab this release https://github.com/fuzeman/Spotify2.bundle/archive/issue/39.zip and install the plugin manually (extract, rename folder to "Spotify2.bundle" and drop into your plugins folder _[might need to restart PMS as well]_).

With this release more debugging information will be logged when you try open that starred playlist again, so you just need provide the latest log file after reproducing the issue again.

**Comment by brumle80** [comment]
tested new zip: http://pastebin.com/xYPJjLPw

**Comment by fuzeman** [comment]
Could you provide a log file after trying to open the "Your Music" -> "Starred" view as well (couldn't see this in that log file)?

Didn't see there was an error logged for the "Playlists" view as well, will have a look at that now:

```
Client Error: Forbidden (403)
```

**Comment by brumle80** [comment]
sorry about that, this should be a test of both playlists and starred, 
albums and artists work now btw :+1: 

http://pastebin.com/sYS3kjEK

also starred out my username, didn't find anything else to block out...

**Comment by fuzeman** [comment]
Could you test out the current development version https://github.com/fuzeman/Spotify2.bundle/archive/develop.zip

This should fix the "Starred" loading bug and will now just exclude playlists from the UI which can't be loaded.

---

I'm not sure why one of your playlists is returning a `Forbidden (403)` response, haven't been able to reproduce this.

We could try find the playlist that's failing, just look for something similar to this in your log file:

```
Error ('Client Error: Forbidden (403)',) returned for request: <HermesRequest uris: ['hm://playlist/user/[username]/playlist/[id]?from=0&length=100']>
```

**Comment by brumle80** [comment]
happy to say it works perfectly :D 
gonna try a few playlists and report if any of them fail, but for now, no issues!
brilliant work :+1:

**Comment by fuzeman** [comment]
That's good, I'll have these fixes included in an update later today.

**Comment by fuzeman** [comment]
These fixes have been released in v0.5.7
